### PR TITLE
Replace creation slot with commitment slot in more places

### LINKED ISSUE #1995: Use commitment input slot or tx creation slot for delegation rewards?
https://github.com/iotaledger/iota-sdk/pull/1961#discussion_r1487367012

### PR DESCRIPTION
**Pull request by thibault-martinez**
Fixes #1995

### CODE REVIEW

File: sdk/src/wallet/operations/transaction/input_selection.rs [hunk 1]
```diff
@@ -125,7 +125,7 @@ where
                     mana_rewards.insert(
                         *output_id,
                         self.client()
-                            .get_output_mana_rewards(output_id, creation_slot)
+                            .get_output_mana_rewards(output_id, slot_commitment_id.slot_index())
```
Comment (Thoralf-M) [inline sdk/src/wallet/operations/transaction/input_selection.rs:128]: This should stay creation_slot, no?

Comment (thibault-martinez) [reply sdk/src/wallet/operations/transaction/input_selection.rs:128]: Idk, Philipp only mentioned potential and decayed mana 🤷🏻‍♂️

Comment (Thoralf-M) [reply sdk/src/wallet/operations/transaction/input_selection.rs:128]: But as exception `The only exception are Potential Mana (generation + decay) and Stored Mana (decay) calculations, which always use the Transaction Creation Slot.`

Comment (thibault-martinez) [reply sdk/src/wallet/operations/transaction/input_selection.rs:128]: But that doesn't mention mana rewards?

Comment (Thoralf-M) [reply sdk/src/wallet/operations/transaction/input_selection.rs:128]: Ah yeah nvm I thought the rewards here are potential mana

**Review by Thoralf-M — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]

**Pull request merged**